File: gems/decomplex/examples/zig/fat-union.zig
pub fn handle(node: Ast) void {
    switch (node) {
        AST.Call => {
            node.line();
            node.col();
            node.ty();
            node.span();
            node.parent();
            node.recv();
        },
        AST.Func => {
            node.line();
            node.col();
            node.ty();
            node.span();
            node.parent();
            node.name();
        },
        AST.Lit => {
            node.line();
            node.col();
            node.ty();
            node.span();
            node.parent();
            node.value();
        },
    }
}

File: gems/decomplex/examples/php/fat-union.php
<?php
function handle($node) {
  switch ($node) {
    case AST::Call:
      $node->line();
      $node->col();
      $node->ty();
      $node->span();
      $node->parent();
      $node->recv();
      break;
    case AST::Func:
      $node->line();
      $node->col();
      $node->ty();
      $node->span();
      $node->parent();
      $node->name();
      break;
    case AST::Lit:
      $node->line();
      $node->col();
      $node->ty();
      $node->span();
      $node->parent();
      $node->value();
      break;
  }
}
